scripts/lib/devtool/ide.py

```diff
@@ -655,8 +664,9 @@ def __vscode_settings_cmake(self, settings_dict):
         """
         if self.build_tool != BuildTool.CMAKE:
             return
-        settings_dict["cmake.configureOnOpen"] = False
+        settings_dict["cmake.configureOnOpen"] = True
         settings_dict["cmake.sourceDirectory"] = self.real_srctree
+        settings_dict["cmake.cmakePath"] = os.path.join(self.staging_bindir_native, "cmake")
```

Comment:
Maybe I should improve the documentation about the cmake-kit use case.

```sh
devtool ide none core-image-minimal

mkdir kit-test
echo "project(foo VERSION 0.1)" > kit-test/CMakeLists.txt 
vscode kit-test
```

Ctrl + Shift + P --> cmake Quick Start --> Select kit

Reply:
Yes, I came to the same conclusion as you. I did check that the preset was correctly using the bitbake CMake, however the kit doesn't. I think I'm not going to focus on the SHARED_SYSROOT_RECIPES for the moment since they are a out of the usual yocto workspace.

We could in the future generate a `.vscode` for the SHARED_SYSROOT_RECIPES in the future. We can already see the limitations of a CMake kit, and this will also be needed for other classes like meson.